### PR TITLE
Reduces allocations in case the tracing is not enabled

### PR DESCRIPTION
**Pull request by MichalPavlik**
Reduces allocations in case the tracing is not enabled

### CODE REVIEW

File: src/Build/BackEnd/Client/MSBuildClient.cs [hunk 1]
```diff
@@ -354,11 +354,11 @@ private bool TrySendPacket(Func<INodePacket> packetResolver)
             {
                 packet = packetResolver();
                 WritePacket(_nodeStream, packet);
-                CommunicationsUtilities.Trace($"Command packet of type '{packet.Type}' sent...");
+                CommunicationsUtilities.Trace("Command packet of type '{0}' sent...", packet.Type);
             }
             catch (Exception ex)
             {
-                CommunicationsUtilities.Trace($"Failed to send command packet of type '{packet?.Type.ToString() ?? "Unknown"}' to server: {0}", ex);
+                CommunicationsUtilities.Trace("Failed to send command packet of type '{0}' to server: {1}", ex, packet?.Type.ToString() ?? "Unknown");
```
Comment (rokonec) [inline src/Build/BackEnd/Client/MSBuildClient.cs:361]: By the format string it looks like parameters shall be reversed. Type shall be 1st and exception 2nd

**Review by rokonec — CHANGES_REQUESTED** [review]

File: src/Build/BackEnd/Client/MSBuildClient.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (AR-May) [reply src/Build/BackEnd/Client/MSBuildClient.cs]: rokonec already mentioned this in his review.

[the remark this answers is not in the document]
Comment (MichalPavlik) [reply src/Build/BackEnd/Client/MSBuildClient.cs]: Looks like a duplicate :)

**Review by AR-May — APPROVED** [review]

File: src/Build/BackEnd/Client/MSBuildClient.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (AR-May) [reply src/Build/BackEnd/Client/MSBuildClient.cs]: yeah, we did simultaneous reviews :)

**Review by rokonec — APPROVED** [review]

**Pull request merged**